vermouth/processors/repair_graph.py

```diff
@@ -194,6 +194,7 @@ def make_reference(mol):
         reference = _get_reference_residue(residues.nodes[residx], mol.force_field)
         if 'mutation' in residues.nodes[residx]:
             resname = residues.nodes[residx]['mutation'][0]
+            residues.nodes[residx]['resname'] = resname
```

Comment:
Technically this works. But wouldn't it be more sane to apply mutations and modifications already in `make residue_graph`. The current fix seems a bit out of place here and it is not really clear, why the statement is there, unless you know that `get_reference_residue` also applies modifications and mutations. Plus if the resname was already change in the residue graph, `_get_reference_residue` would already get the proper residue and you only need to deal with modifications.  

Also do these modifications come from the pdb or the user command line input?

Reply:
I don't disagree that this may not be the best place. But, here's my main arguments against putting it in make_residue_graph: make_residue_graph is used in more places in the codebase, and is, in general, more generally applicable. THerefor it pays to keep that as simple as possible. I'll split it out into new functions though, that should simplify make_reference a bit.

The mutations and modifications come from the CLI at the moment. I guess they could also be added to the PDB parser, if there's sane syntax for it which I doubt.

Reply:
I couldn't find a decent way of splitting it up into smaller functions, so I added a pile of comments instead.